examples/multiple_windows/lib/app/models.dart

```diff
@@ -0,0 +1,116 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: invalid_use_of_internal_member
+// ignore_for_file: implementation_imports
+
+import 'package:flutter/widgets.dart';
+import 'package:flutter/src/widgets/_window.dart';
+
+class KeyedWindow {
+  KeyedWindow({
+    this.parent,
+    this.isMainWindow = false,
+    required this.key,
+    required this.controller,
+  });
+
+  final BaseWindowController? parent;
+  final bool isMainWindow;
+  final UniqueKey key;
+  final BaseWindowController controller;
+}
+
+/// Provides access to the windows created by the application.
+///
+/// The window manager manages a flat list of all of the [BaseWindowController]s
+/// that have been created by the application as well as which controller is
+/// currently selected by the UI.
+class WindowManager extends ChangeNotifier {
+  final List<KeyedWindow> _windows = <KeyedWindow>[];
```

Comment:
There may be a more efficient data structure than a flat list here, but I imagine it will never matter unless someone has an app with a hundred windows for some reason.